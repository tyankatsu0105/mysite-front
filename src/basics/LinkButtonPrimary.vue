<template>
	<g-link
		:to="to"
		class="LinkButtonPrimary"
	>
		{{ text }}
	</g-link>
</template>

<script>
export default {
  props: {
    text: {
      type: String,
      requirer: true,
      default: ""
    },
    to: {
      type: String,
      requirer: true,
      default: "/"
    }
  }
};
</script>

<style scoped lang='scss'>
.LinkButtonPrimary {
  padding: 8px 24px;
  font-size: 1.3rem;
  text-transform: uppercase;
  background-color: $color-black;
  border-radius: 8px;
  box-shadow: $box-shadow-default;
  transition: $tst-default;

  &:hover {
    box-shadow: $box-shadow-default-hover;
  }
}
</style>
