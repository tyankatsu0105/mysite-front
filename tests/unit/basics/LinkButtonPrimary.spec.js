import { shallowMount } from "@vue/test-utils";
import LinkButtonPrimary from "@/basics/LinkButtonPrimary.vue";

const setProps = {
  text: "リンク",
  to: "hogehoge.com"
};

describe("LinkButtonPrimary", () => {
  it("render", () => {
    const wrapper = shallowMount(LinkButtonPrimary, {
      setProps
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
