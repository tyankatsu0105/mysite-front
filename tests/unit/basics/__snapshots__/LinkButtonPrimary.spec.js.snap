// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButtonPrimary render 1`] = `
<g-link class="LinkButtonPrimary">

</g-link>
`;
